cpp/src/arrow/compute/kernels/scalar_string.cc

```diff
@@ -48,6 +48,16 @@ struct AsciiUpper {
   }
 };
 
+struct AsciiLower {
+  template <typename... Ignored>
+  static std::string Call(KernelContext*, const util::string_view& val) {
+    std::string result = val.to_string();
+    std::transform(result.begin(), result.end(), result.begin(),
+                   [](unsigned char c) { return std::tolower(c); });
```

Comment:
Please don't use `std::tolower` as it has locale-dependent behaviour. Instead, just create a hardcoded lookup table.

Reply:
Also, please fix `AsciiUpper` similarly.

Reply:
Note that I used `std::toupper` _just_ for illustration purposes, not because I thought it was the correct way to implement the kernel.